### PR TITLE
Improve logic for building daml-lf-archive-java

### LINKED ISSUE #2058: daml-lf-archive does not contain compiled code
The daml-lf-archive lirary no longer contains compiled code, it only hosts .proto and .java files. It makes this library virtually unusable. The regression happened between versions 100.13.10 and 100.13.11

### PR DESCRIPTION
**Pull request by gerolf-da**
daml-lf-archive-java is put together from the proto sources with the
compiled java classes. However, the logic that picks the java classes
was dependent on the output of $(locations :daml_lf_proto_lib_java),
because it always would pick the first item.

The new logic filters by name rather than position.

Fixes #2058.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by gerolf-da** [comment]
@cocreature: This only showed up in the 0.13.11 release. Is that the first release after the most recent bazel upgrade?

**Comment by cocreature** [comment]
I think so, yeah.

File: daml-lf/archive/BUILD.bazel [hunk 1]
```diff
@@ -195,7 +195,7 @@ genrule(
     ],
     outs = ["daml_lf_archive_java.jar"],
     cmd = """
-    INPUT=$$(echo "$(locations :daml_lf_proto_lib_java)" | cut -d ' ' -f 1)
+    INPUT=$$(echo "$(locations :daml_lf_proto_lib_java)" | tr ' ' '\n' | grep -v '\-src.jar')
```
Comment (remyhaemmerle-da) [inline daml-lf/archive/BUILD.bazel]: Nice !

Comment (gerolf-da) [reply daml-lf/archive/BUILD.bazel]: Please don't endorse me for bash on linkedin 🙏

**Pull request merged**